homeassistant/components/linky/sensor.py

```diff
@@ -1,26 +1,48 @@
 """Support for Linky."""
-import logging
-import json
 from datetime import timedelta
+import json
+import logging
 
+from pylinky.client import DAILY, MONTHLY, YEARLY, LinkyClient, PyLinkyError
 import voluptuous as vol
 
-from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, CONF_TIMEOUT,
-                                 ENERGY_KILO_WATT_HOUR)
 from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.helpers.entity import Entity
-from homeassistant.util import Throttle
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_MONITORED_CONDITIONS, CONF_PASSWORD, CONF_TIMEOUT,
+    CONF_USERNAME, ENERGY_KILO_WATT_HOUR)
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+import homeassistant.util.dt as dt_util
 
 _LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(minutes=10)
+SCAN_INTERVAL = timedelta(hours=4)
+ICON_ENERGY = "mdi:flash"
+CONSUMPTION = "conso"
+TIME = "time"
+INDEX_CURRENT = -1
+INDEX_LAST = -2
+ATTRIBUTION = "Data provided by Enedis"
+
 DEFAULT_TIMEOUT = 10
+SENSORS = {
+    "yesterday": ("Linky yesterday", DAILY, INDEX_LAST),
+    "current_month": ("Linky current month", MONTHLY, INDEX_CURRENT),
+    "last_month": ("Linky last month", MONTHLY, INDEX_LAST),
+    "current_year": ("Linky current year", YEARLY, INDEX_CURRENT),
+    "last_year": ("Linky last year", YEARLY, INDEX_LAST)
+}
+SENSORS_INDEX_LABEL = 0
+SENSORS_INDEX_SCALE = 1
+SENSORS_INDEX_WHEN = 2
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_USERNAME): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
     vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSORS)):
```

Comment:
Per [ADR-003](https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md), we no longer allow the user to configure monitored conditions if all the data is fetched with a single API call (which is the case here). So please remove this option.

Reply:
Done, that's logical.

I'm gonna be more careful about the architecture specs.